### ISSUE TITLE
Hostname Regex Filter

### ISSUE THREAD

**Issue by lampwins** [networktocode/netutils#288]
### Environment
* netutils version:  

### Proposed Functionality

As a Network Engineer I want jinja filter (test) that I can use to evaluate a regex against a device hostname so that I can parse embedded logic within my hostname standards.  

I'd like a helper function in netutils that takes a hostname string and regex as input and returns a boolean based on the regex match. This would then also be exposed as a jinja filter (test) for use in places like Nautobot.

### Use Case

Often I want to make decisions based on the inherent logic comprised in my device hostnames. Most of the time, this results in boolean evaluation of regexs against my hostnames.

**Comment by jdrew82** [comment]
I'm assuming you'd want to parse out the typical things used in a hostname, ie the device site, device role, device position in stack? I'd think maybe passing the regex using named capture groups would be best for that use case.

**Comment by lampwins** [comment]
Yeah I can see two variants here, one, is "does this pattern match or not" and a capture group version.

**Comment by FragmentedPacket** [comment]
Yes both of those would be great.

**Comment by jdrew82** [comment]
I've got this in the capture group form and k:v form in a few SSoT apps that we can pull from. Can we get a list of the expected group names to loom for though? I'm unsure how to do the validation method without another reference of valid values for those groups. Would love to hear thoughts.

**Comment by itdependsnetworks** [comment]
I made this comment [here](https://github.com/networktocode/netutils/pull/289#issuecomment-1585019640)

regex_search, returns a true/false. 
regex_match, returns a list of the capture group. 

The value I see in these is for Jinja, similar to https://netutils.readthedocs.io/en/latest/user/lib_use_cases_jinja_filters/#ipaddress-convenience-functions, and I am not sure the rich object would work, such as `match[1]` after processed through jinja. 

~~~~~~~~~~~~~~~~~~

That being said, I think we could figure a way out for capture group names, but for most of the time, the list returned in should be fine. I think this would solve 80% of the use cases.
